src/plugins/guided_onboarding/server/plugin.ts

```diff
@@ -8,31 +8,51 @@
 
 import { PluginInitializerContext, CoreSetup, Plugin, Logger } from '@kbn/core/server';
 
+import type { GuideId } from '@kbn/guided-onboarding';
 import { GuidedOnboardingPluginSetup, GuidedOnboardingPluginStart } from './types';
 import { defineRoutes } from './routes';
 import { guideStateSavedObjects, pluginStateSavedObjects } from './saved_objects';
+import type { GuideConfig, GuidesConfig } from '../common';
+import { testGuideConfig, testGuideId } from '../common';
 
 export class GuidedOnboardingPlugin
   implements Plugin<GuidedOnboardingPluginSetup, GuidedOnboardingPluginStart>
 {
   private readonly logger: Logger;
+  private readonly guidesConfig: GuidesConfig;
+  private readonly isDevMode: boolean;
 
   constructor(initializerContext: PluginInitializerContext) {
     this.logger = initializerContext.logger.get();
+    this.guidesConfig = {} as GuidesConfig;
+    this.isDevMode = initializerContext.env.mode.dev;
   }
 
   public setup(core: CoreSetup) {
     this.logger.debug('guidedOnboarding: Setup');
     const router = core.http.createRouter();
 
     // Register server side APIs
-    defineRoutes(router);
+    defineRoutes(router, this.guidesConfig);
 
     // register saved objects
     core.savedObjects.registerType(guideStateSavedObjects);
     core.savedObjects.registerType(pluginStateSavedObjects);
 
-    return {};
+    // add a config for a test guide if running in dev mode
+    if (this.isDevMode) {
+      this.guidesConfig[testGuideId] = testGuideConfig;
```

Comment:
Did you consider registering the test guide from the examples plugin instead?

Reply:
Yes, tried that first but there seems to be a known issue with imports to examples from plugins. That means functional tests break if I import `testGuideConfig` into guided onboarding example server side. I decided not to move the config into `kbn-guided-onboarding` package but let me know, if you think it would be best to move it.

Reply:
Ah, I see. I'm OK to leave as-is for now. But maybe let's open up an issue to follow up? I think it'd be preferred to register from the examples plugin, as that's a better representation of how the framework works.